pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -104,6 +104,30 @@ type BMCDetails struct {
 	CredentialsName string `json:"credentialsName"`
 }
 
+// BootMode is the boot mode of the system
+type BootMode string
+
+// Allowed boot mode from metal3
+const (
+	Uefi BootMode = "uefi"
+	Legacy BootMode = "legacy"
+)
+
+// BIOSConfig contains the configuration that you want to configure BIOS settings in Bare metal server
+type BIOSConfig struct {
+
+	// Select the boot mode of the system. Allowed values are uefi, legacy
+	// +kubebuilder:validation:Enum=uefi,legacy
+	BootMode BootMode `json:"bootMode,omitempty"`
```

Comment:
Why mixing boot mode with BIOS configuration? Ironic has a different API for dealing with it (through capabilities), let's make it top level.

Reply:
> Why mixing boot mode with BIOS configuration? Ironic has a different API for dealing with it (through capabilities), let's make it top level.

How does the user of the hardware change the setting directly? It's a BIOS setting, isn't it?

Reply:
I would expect to see something like:

```
 bios:
    bootMode: "uefi"
    irmc:
      enableHyperThreading: true
```

Reply:
> 
> 
> Why mixing boot mode with BIOS configuration? Ironic has a different API for dealing with it (through capabilities), let's make it top level.

Yes, you're right. But some vendors also support configure boot mode via BIOS setting.

Reply:
> 
> 
> I would expect to see something like:
> 
> ```
>  bios:
>     bootMode: "uefi"
>     irmc:
>       enableHyperThreading: true
> ```

Thanks. `bios` will include some common settings and vendor's settings as your suggestion :)

Reply:
I don't know what other folks think, but I think it would make more sense to just have a boolean `legacyMode`.
On a non-x86 architecture it wouldn't make a lot of sense to pass "legacy" and it would definitely make no sense to pass "uefi". It seems less awkward to just have a boolean that is ignored on architectures that don't need it.

Reply:
I agree this should be a top-level item and separate. I don't think it should be a boolean, what if we need to support something else later? It can be omitted in either case.

Reply:
Yeah, moving bootMode out of this struct makes sense to me.